### ISSUE TITLE
Post blocks: ensure all are able to be transformed to other post blocks

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#49283]
All post blocks should be replaceable / convert to other blocks. I should be able to easily swap a Date with an Author block, or Categories with Tags. Currently this is inconsistent and incomplete as you can see below in two examples:

<img width="420" alt="Screen Shot 2023-03-22 at 12 10 02 PM" src="https://user-images.githubusercontent.com/26996883/227011667-cc9f2e29-26d5-4a18-a1a2-4e1a1e212c30.png">

<img width="414" alt="Screen Shot 2023-03-22 at 12 10 07 PM" src="https://user-images.githubusercontent.com/26996883/227011662-4696fd18-b99a-49e4-a5a8-c9f0c11d5f96.png">

Pulled from: https://github.com/WordPress/gutenberg/issues/48443

Marking as needs dev since this seems to need some PRs to ensure transforms are possible.

**Comment by ntsekouras** [comment]
Post terms transforms(categories/tags, etc..) are already in: https://github.com/WordPress/gutenberg/pull/48328.

**Comment by annezazu** [comment]
Awesome! Thanks for noting that.

**Comment by carolinan** [comment]
I think it would be helpful to have a list of the expected transforms, or is it _literally_ "all post blocks should be able to be transformed into all post blocks"

**Comment by mtias** [comment]
Some transforms are somewhat semantic (transforming between content and excerpt, or between different types of post meta), but others are really just about swapping a block with another block.

**Comment by carolinan** [comment]
It is going to be a very long list of blocks in the transform UI, if all post blocks can be transformed to all post blocks:
<img width="323" alt="Transform modal with list of blocks that the excerpt can be transformed to" src="https://user-images.githubusercontent.com/7422055/235554175-744f57b2-2842-4558-94a7-51c749f31056.png">

**Comment by mtias** [comment]
Not necessarily, the transforms list can be for what was explicitly added, and then there can be a "Replace with ->" that opens another menu or quick inserter with access to all the blocks.
